graspologic/match/wrappers.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+from typing import Any, NamedTuple, Optional
+
+import numpy as np
+from beartype import beartype
+from joblib import Parallel, delayed
+from sklearn.utils import check_scalar
+
+from graspologic.match.solver import GraphMatchSolver
+from graspologic.types import Dict, List, RngType
+
+from .types import (
+    AdjacencyMatrix,
+    InitType,
+    Int,
+    MultilayerAdjacency,
+    PaddingType,
+    Scalar,
+)
+
+
+class MatchResult(NamedTuple):
+    indices_A: np.ndarray
+    """
+    Sorted indices in ``A`` which were matched.
+    """
+
+    indices_B: np.ndarray
+    """
+    Indices in ``B`` which were matched. Element ``indices_B[i]`` was matched
+    to element ``indices_A[i]``. ``indices_B`` can also be thought of as a 
+    permutation of the nodes of ``B`` with respect to ``A``.
+    """
+
+    score: float
+    """
+    Objective function value at the end of optimization.
+    """
+
+    misc: List[Dict[str, Any]]
+    """
+    List of length ``n_init`` containing information about each run. Fields for
+    each run are ``score``, ``n_iter``, ``convex_solution``, and ``converged``.
+    """
+
+
+@beartype
+def graph_match(
+    A: MultilayerAdjacency,
+    B: MultilayerAdjacency,
+    AB: Optional[MultilayerAdjacency] = None,
+    BA: Optional[MultilayerAdjacency] = None,
+    S: Optional[AdjacencyMatrix] = None,
+    partial_match: Optional[np.ndarray] = None,
+    init: InitType = "barycenter",
```

Comment:
how about making this `Optional[np.ndarray]` since there's only one Literal we allow?

in otherwords, init=None specifies barycenter? 

since barycenter probably doesnt make sense to most people anyway I think this is better